### PR TITLE
chore: [IOBP-629] Add onboarding payment method when not found any

### PR DESCRIPTION
**Pull request by Hantex9**
## Short description
This PR implements the possibility to add a new payment method during the checkout, being able to resume the payment flow once the method has been successfully added.

**N.B:** ⚠ This behavior has been implemented for a transitional phase with the current PM, as soon as the backend is fully migrated to NPG, this functionality will become deprecated and thus must be removed (reported in the following ticket: https://pagopa.atlassian.net/browse/IOBP-632)

## List of changes proposed in this pull request
- Added a new action to save the `rptId` of the payment in order to resume the transaction later after the onboarding is completed successfully.
- Added a check into [PaymentsOnboardingFeedbackScreen.tsx](https://github.com/pagopa/io-app/compare/master...IOBP-629-add-payment-method-when-not-found-any#diff-9856420745ec55e65f6a52dd2ffc690d46822ac462925108abc93f6a928c4874), if is present a `rptIdToResume` and the outcome is success, It starts the payment flow giving as input the `rptIdToResume`.
- Added an effect into [WalletPaymentPickMethodScreen.tsx](https://github.com/pagopa/io-app/compare/master...IOBP-629-add-payment-method-when-not-found-any#diff-08aa31e861c2cc684eaec44fc714e461c252a9bb06f9d7971429bbdf756af92d) that if the PM doesn't give any payment method onboarded and neither a list of all onboardable payment methods, so it navigates to an outcome screen that shows the CTA to start the onboarding.

## How to test
- Open the new `Payments` section with the feature flag enabled;
- Remove all the payment methods already onboarded;
- Start a new payment flow setting into the `io-dev-api-server` a list of empty payment-methods
- You should be able to onboard a new card and then resume the payment flow

## Preview

https://github.com/pagopa/io-app/assets/34343582/4c524682-61ac-497a-8804-fa7aca3110db

### CODE REVIEW

File: locales/en/index.yml [hunk 1]
```diff
@@ -1224,6 +1224,7 @@ wallet:
         title: "Payment method added successfully!"
         subtitle: ""
         primaryAction: "Vedi dettagli"
+        continueAction: "Continue"
```
Comment (mastro993) [inline locales/en/index.yml:1227]:
```suggestion
        secondaryAction: "Continue"
```

File: locales/en/index.yml [hunk 2]
```diff
@@ -1770,6 +1771,10 @@ wallet:
       METHOD_NOT_ENABLED:
         title: Abilita il metodo per effettuare pagamenti in app
         subtitle: Prima di riprovare con il pagamento, seleziona il metodo nel tuo Portafoglio e abilita la funzionalità “Pagamenti in app”.
+      PAYMENT_METHODS_NOT_AVAILABLE:
+        title: Aggiungi un metodo per effettuare pagamenti in app
+        subtitle: Il metodo verrà salvato nel Portafoglio, così la prossima volta potrai pagare più facilmente.
+        onboardPaymentMethod: Aggiungi metodo
```
Comment (mastro993) [inline locales/en/index.yml:1777]:
```suggestion
        primaryAction: Aggiungi metodo
```
For consistency

File: ts/features/payments/checkout/screens/WalletPaymentPickMethodScreen.tsx [hunk 3]
```diff
@@ -60,13 +64,67 @@ const WalletPaymentPickMethodScreen = () => {
...
+  // If the user doesn't have any onboarded payment method and the backend doesn't return any payment method as guest ..
+  // .. we redirect the user to the outcome screen with an outcome that allow the user to start the onboarding process of a new payment method.
+  // .. This implementation will be removed as soon as the backend will migrate totally to the NPG. (https://pagopa.atlassian.net/browse/IOBP-632)
+  useEffect(() => {
+    if (
+      pot.isSome(paymentMethodsPot) &&
+      _.isEmpty(allPaymentMethods) &&
+      pot.isSome(userWalletsPots) &&
+      _.isEmpty(userPaymentMethods) &&
+      pot.isSome(paymentDetailsPot)
```
Comment (mastro993) [inline ts/features/payments/checkout/screens/WalletPaymentPickMethodScreen.tsx:103]: Can we move all this logic in a selector? For example: `hasValidPaymentMethodsSelector`, which returns `true` or `false` following the logic above.

**Review by mastro993 — CHANGES_REQUESTED** [review]

File: ts/features/payments/checkout/screens/WalletPaymentPickMethodScreen.tsx (hunk 3 above)
Comment (Hantex9) [reply ts/features/payments/checkout/screens/WalletPaymentPickMethodScreen.tsx]: Good idea! Addressed into 20ea0b98f1eab6a7677c8ebba823be0c83ba58b5

File: locales/en/index.yml (hunk 2 above)
Comment (Hantex9) [reply locales/en/index.yml]: Addressed into 20ea0b98f1eab6a7677c8ebba823be0c83ba58b5

File: locales/en/index.yml (hunk 1 above)
Comment (Hantex9) [reply locales/en/index.yml]: Addressed into 20ea0b98f1eab6a7677c8ebba823be0c83ba58b5

**Review by mastro993 — APPROVED** [review]
LGTM! The flow is working as expected!

**Pull request merged**